pkg/util/push/otel.go

```diff
@@ -109,16 +111,44 @@ func OTLPHandler(
 			return body, err
 		}
 
+		log, ctx := spanlogger.New(ctx, "Distributor.OTLPHandler.decodeAndConvert")
```

Comment:
Probably want `spanlogger.NewWithLogger` and use the existing logger from above. Though, the logger from above is still the global logger which isn't great.

Reply:
Good point, fixed in 42eaf65